src/components/pages/Dashboard.tsx

```diff
@@ -1,15 +1,55 @@
 import Link from 'next/link';
+import { useCallback, useEffect, useState } from 'react';
 import { PriceManager } from 'src/components/PriceManager';
 import { Stability } from 'src/components/Stability/Stability';
 import { SystemStats } from 'src/components/SystemStats';
 import { Trove } from 'src/components/Trove/Trove';
 import { Alert } from 'src/components/atoms';
 import { COIN } from 'src/strings';
 import { getFixedIncomeMarketLink } from 'src/utils';
-import { useAccount } from 'wagmi';
+import { useAccount, useWalletClient } from 'wagmi';
+import { useSfStablecoin } from 'src/hooks';
 
 export const Dashboard: React.FC = () => {
-    const { isConnected } = useAccount();
+    const { isConnected, address } = useAccount();
+    const { data: walletClient } = useWalletClient();
+    const [isTokenAdded, setIsTokenAdded] = useState<boolean | null>(null);
+    const {
+        sfStablecoin: {
+            connection: { addresses },
+        },
+    } = useSfStablecoin();
+
+    const checkToken = useCallback(() => {
+        if (!addresses.debtToken) return false;
+        return !!localStorage.getItem(`token_${addresses.debtToken}`);
+    }, [addresses.debtToken]);
+
+    const addToken = useCallback(async () => {
+        if (!walletClient || !address || !addresses.debtToken) return;
+        try {
+            const success = await walletClient.watchAsset({
+                type: 'ERC20',
+                options: {
+                    address: addresses.debtToken,
+                    symbol: COIN,
+                    decimals: 18,
+                    image: 'https://app.usdfc.net/apple-touch-icon.png',
+                },
+            });
+            if (success) {
+                localStorage.setItem(`token_${addresses.debtToken}`, 'true');
+                setIsTokenAdded(true);
+            }
+        } catch {}
+    }, [walletClient, address, addresses.debtToken]);
```

Comment:
In `SystemStats`, there is a same hook. Both should store the flag on the local storage.
So let's make this a common hook and put under `src/hooks` and use this at the both components.

Also, in this `try/catch`  there is no code for error handling.
It means you can't notice when an error happens.
So you should remove the `try/catch`.

Reply:
I didn't mean both solutions are needed.
Let's remove 1st one and leave a close button.